src/test/java/de/retest/suite/flow/CreateChangesetForAllDiffrencesFlowTest.java

```diff
@@ -0,0 +1,100 @@
+	private static final String ACTION = "launch";
+	private static final String FILE = "src/test/resources/recheck/" + SUITE + "/" + TEST + "." + ACTION + ".recheck";
+
+	@Rule
+	public TemporaryFolder temp = new TemporaryFolder();
+
+	@Rule
+	public SystemProperty recheckDirProp = new SystemProperty( ProjectConfiguration.RETEST_PROJECT_ROOT );
+
+	@Before
+	public void setUp() throws Exception {
+		recheckDirProp.setValue( temp.getRoot().getAbsolutePath().toString() );
+		Files.createDirectories( temp.getRoot().toPath().resolve( "src/test/java" ) );
+	}
+
+	@Test
+	public void apply_should_throw_exception_whitout_the_stateFilePath() throws Exception {
+		final PersistenceFactory persistenceFactory =
+				new PersistenceFactory( new HashSet<>( Arrays.asList( StdXmlClassesProvider.getXmlDataClasses() ) ) );
+
+		final Persistence<SutState> persistence = persistenceFactory.getPersistence();
+		final File originalFile = new File( FILE );
+		final File temporaryFile = temp.newFile( "check_GUI_with_review_license.launch.recheck" );
+
+		FileUtils.copyFile( originalFile, temporaryFile );
+		final SutState sutState = persistence.load( temporaryFile.toURI() );
+
+		final ReviewResult reviewResult = new ReviewResult();
+		final SuiteChangeSet changesWhitPathAndDescription = reviewResult.createSuiteChangeSet( SUITE, "" );
+		final SuiteChangeSet changesWhitoutAnything = reviewResult.createSuiteChangeSet( SUITE, "" );
+		final SuiteChangeSet changesWhitoutStateFilePath = reviewResult.createSuiteChangeSet( SUITE, "" );
+		final SuiteChangeSet changesWhitoutDescription = reviewResult.createSuiteChangeSet( SUITE, "" );
+
+		final Element toChange = sutState.getRootElements().get( 0 ).getContainedElements().get( 0 )
+				.getContainedElements().get( 0 ).getContainedElements().get( 0 ).getContainedElements().get( 0 );
+
+		changesWhitPathAndDescription.createTestChangeSet().createActionChangeSet( ACTION, temporaryFile.getName() )
+				.getAttributesChanges()
+				.add( toChange.getIdentifyingAttributes(), new AttributeDifference( "enabled", "false", "true" ) );
+
+		changesWhitoutDescription.createTestChangeSet().createActionChangeSet( null, temporaryFile.getName() )
+				.getAttributesChanges()
+				.add( toChange.getIdentifyingAttributes(), new AttributeDifference( "enabled", "false", "true" ) );
+
+		//if createActionChangeSet is called whitout the stateFilePath, a NoStateFileFoundException is thrown
+		changesWhitoutStateFilePath.createTestChangeSet().createActionChangeSet( ACTION, null ).getAttributesChanges()
+				.add( toChange.getIdentifyingAttributes(), new AttributeDifference( "enabled", "false", "true" ) );
+
+		changesWhitoutAnything.createTestChangeSet().createActionChangeSet().getAttributesChanges()
+				.add( toChange.getIdentifyingAttributes(), new AttributeDifference( "enabled", "false", "true" ) );
+
+		assertThatCode( () -> ApplyChangesToStatesFlow.apply( persistence, changesWhitPathAndDescription ) )
+				.doesNotThrowAnyException();
```

Comment:
We usually don't use this assertion because if a particular statement throws an exception, the test fails anyway. However, is this critical? I don't know the story behind it.

Reply:
Jeremy had a problem that this NoStateFileFoundException was thrown, when he had tested something. He solved it with the other constructor, where the 2 parameters are passed, but he could not verify to 100% that this will fix the problem.
So I wanted to prove that the exception is not thrown when stateFilePath is passed.